lib/kuzzle/kuzzle.js

```diff
@@ -55,47 +54,53 @@ const InternalIndexHandler = require('./internalIndexHandler');
  */
 class Kuzzle extends KuzzleEventEmitter {
   constructor (config) {
+    if (global.kuzzle) {
+      throw new Error('Cannot build a Kuzzle instance: another one already exists');
+    }
+
     super(
       config.plugins.common.maxConcurrentPipes,
       config.plugins.common.pipesBufferSize);
 
+    Reflect.defineProperty(global, 'kuzzle', {
+      value: this,
+    });
+
```

Comment:
I don't understand why a `global` level is needed.
Can't we get rid of it and simply write `kuzzle.something` instead of `global.kuzzle.something` ?
With the `global` It feels like there are others globals
Sorry it its a naive question

Reply:
I don't get it. Where do you want to put it?

I had another approach with a static instance getter. It also works, and it only needs a `require` from files needing it. This approach has two caveats: it's really inconvenient for our private plugins, and it's a pain in the a** to mock (I succeeded only by monkey patching mock-require :expressionless: )

But apart from this other solution, I don't see where I can store that variable.